processor/groupbyattrsprocessor/attribute_groups.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groupbyattrsprocessor
+
+import (
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// attributeEntry keeps a grouping attributes and matching records
+type attributeEntry struct {
+	attrs         pdata.AttributeMap
+	resourceSpans pdata.ResourceSpans
+	resourceLogs  pdata.ResourceLogs
+}
+
+func instrumentationLibrariesEqual(il1, il2 pdata.InstrumentationLibrary) bool {
+	return il1.Name() == il2.Name() && il1.Version() == il2.Version()
+}
+
+// instrumentationLibrarySpans searches for a pdata.InstrumentationLibrarySpans instance matching
+// given InstrumentatonLibrary. If nothing is found, it creates a new one
+func (ae *attributeEntry) instrumentationLibrarySpans(library pdata.InstrumentationLibrary) pdata.InstrumentationLibrarySpans {
+	ilss := ae.resourceSpans.InstrumentationLibrarySpans()
+	for i := 0; i < ilss.Len(); i++ {
+		ill := ilss.At(i)
+		if instrumentationLibrariesEqual(ill.InstrumentationLibrary(), library) {
+			return ill
+		}
+	}
```

Comment:
Since operator == compares pointers - this isn't a working approach. Then either it can be applied to the struct with 2 fields  - name and version - which isn't good since the structure may be changed or actually leave it how it is done now